SearchActivity.py

```diff
@@ -206,28 +209,28 @@ def _joined_cb(self, activity):
 
     def _new_tube_common(self, sharer):
         """ Joining and sharing are mostly the same... """
-        if self._shared_activity is None:
+        if self.get_shared_activity() is None:
```

Comment:
Is this change necessary? Can this change might be a reason for the port not to work properly? @quozl @tonadev .

Reply:
`self._shared_activity` is in the GTK 2 toolkit _sugar-toolkit_, and is not in the GTK 3 toolkit _sugar-toolkit-gtk3_.

`self.get_shared_activity` is in both toolkits.

My guess is that the code hadn't been changed during the port to GTK 3, before the port to TelepathyGLib.  If we're not sure if collaboration worked for the GTK 2 release of the activity, the way to be sure is to test that release.

Reply:
Yes. I also think that collaboration should be tested for GTK 2 first, It seems that there are not much changes from telepathy to TelepathyGlib for this activity, and the differences @quozl pointed out might be seen before telepathy port.

> Game playing board state is mostly shared, but not fully; differences are;

Now seen a couple of instances where collaboration code is not properly ported to GTK 3 .

Reply:
Yes, it is endemic.  Root cause is missing-in-action activity maintainers, and well-meaning volunteers during GCI or students in GSoC who have concentrated on a GTK 3 port without fully understanding the collaboration feature, and reviewers with the same limited focus.  Another cause is activities copied from other activities that do collaboration, without implementing collaboration in the new activity.  Is why we re-worded the project idea this time, and @nswarup14 is doing "Maintain 25 activities" rather than "Port to GTK 3".